Linting randomly breaks with 4.28.0

Plugin gets stuck highlighting non-errors in code, need to close/re-open file as a workaround.

![Image](https://github.com/user-attachments/assets/ab1516df-dbf1-40e2-bae1-ced7b22f4ff1)

![Image](https://github.com/user-attachments/assets/4594b3c9-b49b-4dc3-84a9-ade973df5588)

Here I give an example, while I intentionally edit the type in the message and change it to int to cause the error, I go back to correct it to str but then you can see the mypy show:

```
 28:36  error  mypy    invalid syntax 
```

--- a/lint/linter.py
+++ b/lint/linter.py
@@ -1115,7 +1115,11 @@ class Linter(metaclass=LinterMeta):
             return True
 
         effective_lint_mode = get_effective_lint_mode(settings)
-        if real_file_linter and "on_modified" in effective_lint_mode:
+        # Treat a save as a valid trigger when the user opted into
+        # on_modified. This matches the semantic intent that
+        # "on_modified" means "run on edits", and a save after edits
+        # should not suppress a pending lint.
+        if "on_modified" in effective_lint_mode:
             effective_lint_mode.add("on_save")
         ok = reason in effective_lint_mode
 

--- a/tests/test_sublime_linter_entrypoint.py
+++ b/tests/test_sublime_linter_entrypoint.py
@@ -75,7 +75,7 @@ class TestLinterElection(_BaseTestCase):
         verify(backend.orchestrator).submit(...)
 
     @p.expand([
-        (reason, lint_mode, ok)
+        (f"{lint_mode}-{reason}-{ok}", reason, lint_mode, ok)
         for lint_mode, reasons in {
             'background': [
                 ('on_user_request', True),
@@ -108,7 +108,7 @@ class TestLinterElection(_BaseTestCase):
             ('on_modified',): [
                 ('on_user_request', True),
                 ('config_changed', True),
-                ('on_save', False),
+                ('on_save', True),
                 ('on_load', False),
                 ('on_modified', True),
             ],
@@ -129,7 +129,7 @@ class TestLinterElection(_BaseTestCase):
             ('on_modified', 'on_load'): [
                 ('on_user_request', True),
                 ('config_changed', True),
-                ('on_save', False),
+                ('on_save', True),
                 ('on_load', True),
                 ('on_modified', True),
             ],
@@ -157,7 +157,7 @@ class TestLinterElection(_BaseTestCase):
         }.items()
         for reason, ok in reasons
     ])
-    def test_background_capable_linter_matches_reason_and_lint_mode(self, reason, lint_mode, ok):
+    def test_background_capable_linter_matches_reason_and_lint_mode(self, _, reason, lint_mode, ok):
         class FakeLinter(Linter):
             defaults = {'selector': '', 'lint_mode': lint_mode}
             cmd = 'fake_linter_1'

